url4192078a2f_unsegnor_hipotetizador_tgz-pJ8-hipotetizador_MedirTiemposJ8/src/hipotetizador/TDFPG.java

```diff
@@ -0,0 +1,593 @@
+/*
+ * To change this template, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package hipotetizador;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+
+/**
+ * Top Down Frequent Pattern Growth
+ *
+ * @author Víctor
+ */
+public class TDFPG {
+
+    
+
+    
+    
+    /**
+     * Contiene el Elemento, su frecuencia y el puntero al primer nodo
+     */
+    //ArrayList<RegistroTD> tabla = new ArrayList< >();
+    //Esta función espera la lista de elementos filtrados por soporte mínimo
+    //VOY A DESCOMPONERLA!!
+    /**
+     * 
+     * @param tabla_frecuencias
+     * @param historia
+     * @param tventana
+     * @param generar_todas Si está a true indica que se generen también las subreglas
+     * @return 
+     */
+    public ArrayList<Regla> extraer_reglas(ArrayList<InfoElemento> tabla_frecuencias, boolean[][] historia, int tventana, boolean generar_todas) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This method contains 45 lines of code, not including blank lines or lines with only comments, Java punctuation characters, identifiers, or literals. By comparison, 98% of the methods in the CodeGuru reference dataset contain fewer lines of code. Large methods might be difficult to read and have logic that is hard to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 45-62 into a separate method.